modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.js

```diff
@@ -95,10 +95,12 @@ export default class ScenegraphLayer extends Layer {
     }
 
     const {data, getPosition} = this.props;
-    const {value} = attribute;
-    let i = 0;
-    for (const point of data) {
-      const position = getPosition(point);
+    const {value, size} = attribute;
+    let i = startRow * size;
+    const {iterable, objectInfo} = createIterable(data, startRow, endRow);
+    for (const point of iterable) {
+      objectInfo.index++;
```

Comment:
Just a refresher: What is the `objectInfo.index++` statement doing? Why does it need to be done outside of the iterator? Is it because custom geometry layers need to control it?

Reply:
We found that creating a custom iterable object is magnitudes slower than iterating through a native array. So `iterable` here is always an array, and we manually maintain an `obectInfo` object that contains the contextual information.